components/JbrowseCircular.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {
+  createViewState,
+  JBrowseCircularGenomeView,
+} from "@jbrowse/react-circular-genome-view";
+import { useEffect, useState } from "react";
+import { assembly } from "@/jbrowse/assembly";
+import { CircularViewModel, JbrowseFileInput } from "./types";
+import { getTracks } from "./common";
+import { circularTracks } from "@/jbrowse/circular/tracks";
+import { defaultCircularOptions } from "@/jbrowse/circular/dynamic";
+
+export const JbrowseCircular = ({
```

Comment:
circular version of our jbrowse component

- it accepts a list of selected file IDs and creates the tracks (in the final version this would fetch song URLs)
- it also accepts an object `options` that is passed to createViewState.